Lib/ast.py

```diff
@@ -52,6 +52,8 @@ def _convert_num(node):
                 return node.value
         elif isinstance(node, Num):
             return node.n
+        elif isinstance(node, AST):
+            raise ValueError('%s not allowed in literal' % type(node).__name__)
         raise ValueError('malformed node or string: ' + repr(node))
```

Comment:
In what cases this exception is raised? Is it covered by tests?

Reply:
The easiest way to trigger it is to use something that isn't permitted in literal_eval, but is valid syntax. There's already one such test - proving that f-strings aren't literal_evallable - and I guess there could be more, but since this is a whitelist, the error is basically "anything else, raise". The only difference in this patch is that the message is different for "syntactically valid but not acceptable by literal_eval" and "syntactically invalid".

Reply:
I meant what input triggers the "malformed node or string" error now?

Reply:
Calling literal_eval with an AST node that has non-AST children in it. I don't think it's possible (post-patch) to trigger this error by calling literal_eval with a string, but I can't say for sure, so I don't want to say just "malformed node".